From dc04457a26de6ec01de9c9f9bfcaf1a0dfe4be15 Mon Sep 17 00:00:00 2001
From: "xinhui.pan" <xinhuix.pan@intel.com>
Date: Tue, 14 Apr 2015 16:55:59 +0800
Subject: [PATCH] Touch: free devm resource correctly in probe

use devm_kfree instead of kfree, otherwise panic will hit.

Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-33182
Change-Id: I62f60bcc3e2dfd847d4f32722d652beb0e727415
Signed-off-by: xinhui.pan <xinhuix.pan@intel.com>
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index d481db8..b47c5e7 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -3894,7 +3894,7 @@ err_free_irq:
 	free_irq(client->irq, data);
 err_free_mem:
 	if (data->alloc_pdata) {
-		kfree(data->pdata);
+		devm_kfree(&client->dev, data->pdata);
 		kfree(data->fw_name);
 		kfree(data->cfg_name);
 	}
@@ -3930,7 +3930,7 @@ static int mxt_remove(struct i2c_client *client)
 	regulator_put(data->reg_vdd);
 	mxt_free_object_table(data);
 	if (data->alloc_pdata) {
-		kfree(data->pdata);
+		devm_kfree(&client->dev, data->pdata);
 		kfree(data->fw_name);
 		kfree(data->cfg_name);
 	}
-- 
1.9.1

